src/mongo/db/curop.h

```diff
@@ -305,6 +306,10 @@ class CurOp {
         if (_dbprofile <= 0)
             return false;
 
+        if (serverGlobalParams.rateLimit > 1 && _dbprofile >= 2 && ms < serverGlobalParams.slowMS) {
```

Comment:
My idea is to leave logic unchanged when rateLimit is 1 because 1 means no events are filtered. I believe when rateLimit is 1 this function behaves exactly the same way as it does in upstream (without rate limiter feature).